ArtOfIllusion/src/artofillusion/material/LinearMaterialMapping.java

```diff
@@ -45,7 +45,8 @@ public String getName()
     return "Linear";
   }
 
-  public static boolean legalMapping(Object3D obj, Material mat)
+  @Override
+  public boolean legalMapping(Object3D obj, Material mat)
```

Comment:
I _think_ that this is part of trying to get out of using reflection. Instead of `instance.getClass().getMethod("legalMapping").invoke(obj, mat)` we would call `instance.legalMapping(obj, mat)` directly.  For this to work, the method must not be static, as static methods get called based in the _reference type_ (ie, we would always get what is defined in `MaterialMapping`, even if what we really have is a `LinearMaterialMapping`)

Regarding use in Plugins, It does not appear that any of the current major plugins create custom mappings. Polymesh has a class that __should__ be, but is defined outside the inheritance tree...

I have a thought in mind that _would_ want to use this as an extension point, but it does not exist yet.

All in all, I do think that this way is cleaner. I'm not sure why reflection was being used here in the first place. Even the `PluginRegistry` actual object instances to work from.